src/xtgeo/interfaces/rms/rmsapi_utils.py

```diff
@@ -0,0 +1,442 @@
+
+
+class _StorageTypeRegularSurface(str, enum.Enum):
+    """Valid 'storage' types for reg. surf in RMS, invoked with 'stype=...'"""
+
+    HORIZONS = "horizons"
+    ZONES = "zones"
+    CLIPBOARD = "clipboard"
+    GENERAL2D_DATA = "general2d_data"
+    TRENDS = "trends"
+
+    @classmethod
+    def values(cls) -> list[str]:
+        """Return list of all valid surface type values."""
+        return [item.value for item in cls]
+
+
+class RmsApiUtils:
+    """Class RmsApiUtils, for accessing project level methods::
+
+     import xtgeo
+
+     xr = xtgeo.RmsApiUtils(project)
+     xr.create_horizon_category('DS_extracted_run3')
+     xr.delete_horizon_category('DS_extracted_run2')
+
+    The project itself can be a reference to an existing project, typically
+    the magic ``project`` wording inside RMS python,
+    or a file path to a RMS project (for external access).
+
+    Args:
+        project (rmsapi.Project or str): Reference to a RMS project
+            either an existing instance or a RMS project folder path.
+        readonly (bool). Default is False. If readonly, then it cannot be
+            saved to this project (which is the case for "secondary" projects).
+
+    Examples::
+
+        import xtgeo
+        path = '/some/path/to/rmsproject.rmsx'
+
+        ext = xtgeo.RoxUtils(path, readonly=True)
+        # ...do something
+        ext.safe_close()
+
+    """
+
+    def __init__(self, project: RmsProjectType, readonly: bool = False) -> None:
+        self._project = None
+
+        if rmsapi is None:
+            raise RuntimeError(
+                "rmsapi package is not available. "
+                "Please install it to use RMS features."
+            )
+
+        # After this check, rmsapi is guaranteed to be not None
+        assert rmsapi is not None  # help mypy understand this
+
+        self._version = rmsapi.__version__
+
+        if versionparse(self._version) < versionparse("1.10"):
+            raise RuntimeError("XTGeo >= 4.0 requires rmsapi API >= 1.10")
+
+        self._rmsexternal = True
+
+        self._versions = {
+            "1.0": ["10.0.x"],
+            "1.1": ["10.1.0", "10.1.1", "10.1.2"],
+            "1.1.1": ["10.1.3"],
+            "1.2": ["11.0.0"],
+            "1.2.1": ["11.0.1"],
+            "1.3": ["11.1.0", "11.1.1", "11.1.2"],
+            "1.4": ["12.0.0", "12.0.1", "12.0.2"],
+            "1.5": ["12.1"],
+            "1.6": ["13.0"],
+            "1.7": ["13.1"],
+            "1.7.1": ["13.1.1"],
+            "1.7.2": ["13.1.2"],
+            "1.8": ["14.0", "14.0.1"],
+            "1.9": ["14.1"],
+            "1.10": ["14.2"],
+            "1.11": ["14.2.1", "14.2.2"],
+            "1.12": ["14.5"],
+            "1.12.1": ["14.5.0.1"],
+            "1.13": ["15", "15.0.1.0"],
+        }
+
+        if project is not None and isinstance(project, (str, pathlib.Path)):
+            projectname = str(project)
+            assert rmsapi is not None  # guaranteed by __init__
```

Comment:
Is mypy still complaining down here, could this together with the assert on line 101 be removed? 
I thought ` if rmsapi is None:` on line 94 should be enough for mypy (but mypy has surprised me before 😄)

Reply:
You are right; I can remove this assert. I cannot even recall why I needed it at some point...